### ISSUE TITLE
sqlalchemy

### ISSUE THREAD

**Issue by chadwhitacre** [gratipay/team-review#11]
https://gratipay.com/sqlalchemy/

**Comment by chadwhitacre** [comment]
From @zzzeek at https://github.com/gratipay/inside.gratipay.com/issues/234#issuecomment-129045466:

> @whit537 re: "it seems that SQLAlchemy has a pretty well-established funding model outside of Gratipay", I'm not really sure what that is based on.  
> 
> SQLAlchemy's "funding model" is based on:
> 1. I work for a salary full time
> 2. I pay for all SQLAlchemy hosting expenses directly from personal funds, answer 90% of all mailing list issues, write 95% of all code, do 100% of all releases, bug triage, web site maintenance, etc.
> 
> and...that's it!   There are almost entirely _no_ lines of code in SQLAlchemy that were paid for by _anyone_, I can point you to the two things that were paid: the original version of the [versioning example](http://docs.sqlalchemy.org/en/rel_1_0/orm/examples.html#module-examples.versioned_history) was paid for, and a [transaction replay extension](https://bitbucket.org/zzzeek/sqlalchemy/issues/3104/transaction-replay-extension) which has never been released was paid for.  **Everything else you see in SQLAlchemy, code, docs, examples, bugfixes, tests was written for free**.
> 
> Now to be fair, my current employer is a large open source company, so as of the past year, I finally can spend a small portion of my time working on SQLAlchemy and Alembic issues as part of my job.    But I still have to run and fund the project entirely myself.
> 
> I'm very excited for the moment by a new Bountysource platform that offers recurring payments, as open source projects like SQLAlchemy IMO deserve to be allowed to receive monthly funding from supportive users without passing through arcane ideological hoops.

**Comment by mattbk** [comment]
Looks good to me.  There is a guide for getting involved at http://www.sqlalchemy.org/develop.html.

**Comment by chadwhitacre** [comment]
More comments from https://github.com/gratipay/inside.gratipay.com/issues/234 for context:

@rohitpaulk:

> > How Revenue Is Shared
> > Contributors are paid directly via the customers they support on a contract or permanent basis.
> 
> What about Gratipay revenue?

@webmaven:

> SQLAlchemy is an interesting case. It is an open source project, whose development has been driven greatly by for-profit consulting (by many parties) to add enhancements. It isn't really structured to share donations among voluntary contributors from it's community (which is largely steered towards bug reporting and answering Qs on the mailing list and IRC channel), and I think the answer here indicates that [Authors](http://www.sqlalchemy.org/develop.html#roster) are generally assumed to be contributing patches that solve a problem for themselves (or for their clients). I could be wrong, but that seems to mean that all donations via PayPal are currently going directly to @zzzeek.

me:

> @zzzeek Echoing @webmaven's comments above at #234 (comment), it seems that SQLAlchemy has a pretty well-established funding model outside of Gratipay, and the question for us here is how to model or adapt that for SQLAlchemy's presence on Gratipay. Both the revenue model and revenue sharing sections describe activity out-of-band from Gratipay's perspective. What about revenue generation and sharing on Gratipay?

**Comment by chadwhitacre** [comment]
> I'm very excited for the moment by a new Bountysource platform that offers recurring payments, as open source projects like SQLAlchemy IMO deserve to be allowed to receive monthly funding from supportive users without passing through arcane ideological hoops.

@zeek Okay. This makes it sound like you're no longer interested in using Gratipay, so that the best thing to do would be to ~~reject~~[close](https://github.com/gratipay/gratipay.com/pull/3678) your Team application.

Objections, anyone?

**Comment by mattbk** [comment]
There's nothing requiring revenue-sharing _through_ Gratipay in the terms, so why is that being used as criteria for rejection?  There isn't even a mechanism for this because payroll is still broken.

It's kind of ridiculous that @zzzeek has been waiting over a month for a decision on his application.  This indicates to me that the criteria for acceptance or rejection are unclear.

**Comment by chadwhitacre** [comment]
> It's kind of ridiculous that @zzzeek has been waiting over a month for a decision on his application.

We were waiting to hear back from him. He didn't address any of our [comments](https://github.com/gratipay/applications/issues/11#issuecomment-130410465) until last week.

> There's nothing requiring revenue-sharing _through_ Gratipay in the terms, so why is that being used as criteria for rejection?

Is it? I interpreted @zzzeek's comment as a withdrawal of his application, no?

**Comment by zzzeek** [comment]
i don't really understand why gratipay need have such a complex model.     OSS developers want to receive monthly contributions for their work.  We sign up, people can send us money.  Paypal, Flattr, Patreon, and endless other services exist that let people give money to other people.   OSS developers need to get paid for their work, gittip was great for this, but with gratipay, it's just making things hard for us again.  that frustrates me and I think it will generally frustrate others as well.

**Comment by chadwhitacre** [comment]
> i don't really understand why gratipay need have such a complex model.

@zzzeek Gratipay needs to have a complex model because we're doing a hard thing.

We're trying to implement open culture as an abstraction atop the existing global legal and financial system.

We're trying to cultivate an economy of gratitude, generosity, and love, by offering voluntary recurring payments and payroll for open work.

First, as to payments, open-source projects lack income for lack of (a) savvy and (b) a culture of voluntary payments, but not for lack of (c) tools. As you point out, there are plenty of tools—PayPal, Flattr, Patreon, Bountysource. But as to savvy: What's your revenue model? Who are your customers, and what's your process for converting them into (voluntarily) _paying_ customers via whatever tool(s) you prefer? This is a business question, not a technology question. The cultural question _is_ related to tooling in that better tools amplify cultural change through their network effects, and also to savvy in that each project's education of their own customers also amplifies the network effect. We had a network effect going for a time with early Gittip, but we lost our momentum because of our individual-centric model and our naive approach to community management.

SQLAlchemy to date has been primarily funded by your salary. This is a time-honored way to fund open-source software, but it's not what Gratipay (at least) is offering. We're offering a _new_ way to fund open-source software which is more akin to a business, where your customers pay you for what you provide. Gratipay's ideal customer is an _open company_. We're not targeting open-source projects _per se_, though of course it's possible to run an open-source project as an open company.

The trick is that an open company's customers pay _after_ you provide your product, and all of us who are trying to find a way to make this model work have to find a way to convert customers into _voluntarily paying_ customers. Wikipedia has by far been the most successful so far, and for Gratipay the most successful thing we've done is to [add a site-wide call-to-action](http://blog.gittip.com/post/95126362826/we-depend-on-voluntary-payments/) similar to how Wikipedia does it. SQLAlchemy has a donation link [on the site](http://www.sqlalchemy.org/). Is that working? Why or why not? What else could you do? Do you include calls to action with all of your releases? Do you include calls to action in your mailing list signature? Do you have a sales process for reaching out to regular customers? Etc.

If you're comfortable with the status quo for SQLAlchemy as an open-source project, wherein you and other contributors find day jobs that give you more or less flexibility to build SQLAlchemy, then neither Gratipay nor PayPal nor Flattr nor Patreon nor Bountysource have much to offer you. We'll always be at most a "bonus" for you. Gittip may have been a nice bonus for you in particular, but we want to provide more than just a bonus for a lucky few, and we didn't think we could achieve that without the Gratipay 2.0 refactor. Sorry. :-(

If you want to turn SQLAlchemy into a business that is funded more or less entirely by voluntary payments from your customers, then in our view this requires a different mindset, and that's what the "revenue model" question on our Team application is designed to suss out.

Now, I agree that purely from a usability point of view, Gratipay kind of sucks right now, and worse than usual because of all the [disruption](https://medium.com/gratipay-blog/off-balanced-e3d19f2abece) of the Balanced shutdown. Even apart from that, we lack resources, because we're strongly committed to open culture (so no venture capital) and to our own platform (so no out-of-band grants). That means we're going to watch more Patreons hockeystick while we will grow gradually at best. Maybe Bountysource will completely eat our open-source lunch after all. Since we're barely usable right now, I can't blame you for preferring them to us. I don't know what to do about this except to keep chipping away on PRs. :disappointed:

Second, payments is only half the problem. To turn an open-source project into a full-fledged open company (which is what we're looking for), one has to solve the _payroll_ problem. Benjamin Mako Hill talks about this extensively in "[Problems and Strategies in Financing Voluntary Free Software Projects](https://mako.cc/writing/funding_volunteers/funding_volunteers.html)," which we've been [picking apart](https://github.com/gratipay/inside.gratipay.com/issues/179). Here's the key premise:

> It's easier for a successful volunteer Free Software project to get money than it is to decide how to spend it.

The team take model that we discovered with Gratipay 1.0 has shown "[exciting promise](https://medium.com/gratipay-blog/gratipay-2-0-2453d3c53077)" as an answer to this problem, and that's where we're going to be putting our energy with Gratipay 2.0. It's a pretty hard problem, because on the one hand it gets tangled up in [employment law](https://github.com/gratipay/inside.gratipay.com/issues/242) and [storing PII](https://github.com/gratipay/gratipay.com/issues/3504), and on the other hand we have to keep it simple enough that we don't lose too much of [the simplicity of the loose associations](https://github.com/gratipay/gratipay.com/issues/3556#issuecomment-126561186) that have historically characterized open-source. The question we're out to answer is whether this equation can balance:

`open-source project + money = open company`

# tl;dr

Gratipay needs to have a complex model because we're doing a hard thing.

**Comment by chadwhitacre** [comment]
I mean, we're not _trying_ to make Gratipay complex, it's just that making complicated things simple is hard work. :-(

**Comment by zzzeek** [comment]
> Who are your customers, and what's your process for converting them into (voluntarily) paying customers via whatever tool(s) you prefer? 

you need to just create a culture where companies that have the cash are _expected_ to do so.  look at all the logos on https://us.pycon.org/2015/.   Why do they do that?  Just so they can get their logo there and be seen as "supporting open source".   Yeah, they get a booth for two days too, big whoop.  They want their logo up there.   This was what I had in mind for Gittip early on.     This is what paypal, flattr, and patreon are _not_ doing; marketing themselves towards open source software as the _one thing they do_.   Gittip was the only company doing this.

> Wikipedia has by far been the most successful so far, and for Gratipay the most successful thing we've done is to add a site-wide call-to-action similar to how Wikipedia does it. 

OK, I guess try to do what they're doing but Wikipedia is an enormous organization and their business is not writing software, it's crowdsourcing content that is universally valuable to virtually every human being on the planet (so the number of people who directly see benefit in their lives from Wikipedia is probably over _a billion people_).  I see little similarity between them and your average open source software project that is valuable to the no more then five-ten thousand other programmers who find it useful.

> If you're comfortable with the status quo for SQLAlchemy as an open-source project, wherein you and other contributors find day jobs that give you more or less flexibility to build SQLAlchemy, then neither Gratipay nor PayPal nor Flattr nor Patreon nor Bountysource have much to offer you. 

We will see about Bountysource, there's only ten projects [listed](https://salt.bountysource.com/) on their new monthly site and the Neovim project, certainly a tool that's at least half as popular as Wikipedia (by which I mean, WTF is Neovim and do what, 10000 people in the world use it maybe?) is somehow making $3K a month.  

> We'll always be at most a "bonus" for you.

Well the site was named "Gittip", which I know you had a weird pronunciation with, but most of us thought "tip" meant "gratuity".  E.g. a bonus.  Seemed like a worthy idea in its own, if not a system of providing full time salaries to people.

> If you want to turn SQLAlchemy into a business that is funded more or less entirely by voluntary payments from your customers, then in our view this requires a different mindset, and that's what the "revenue model" question on our Team application is designed to suss out.

OK let me make an important point about this.  _Of course_ we all want our OSS projects to be entirely self-sustaining entities that we make our living on.  But if gratipay's model is based on, "you need to go full time on your OSS project as a business, and make no money from anywhere else like a dayjob in order to sign up for us", that's not really a reasonable ask.   Most higher quality OSS projects are run by older folks who have dayjobs, families, mortgages who can't responsibly just chuck the whole thing to go full blown independent OSS vendor overnight.   Most OSS projects are built by people who have day jobs.   It seems like the model you're pursuing here has discrimination against people with day jobs built right into it.  

> It's easier for a successful volunteer Free Software project to get money than it is to decide how to spend it.

That's an absurd statement.  I can easily decide how to spend money that is donated to me in return for the many years of work I've put into SQLAlchemy and into helping individual users to get their software working.   I would use it to feed my family.

**Comment by chadwhitacre** [comment]
> OK, I guess try to do what they're doing but Wikipedia is an enormous organization and their business is not writing software, it's crowdsourcing content that is universally valuable to virtually every human being on the planet (so the number of people who directly see benefit in their lives from Wikipedia is probably over _a billion people_).

... and their [revenue](https://meta.wikimedia.org/wiki/Wikimedia_Foundation_Annual_Plan/2015-16#/media/File:Screen_Shot_2015-06-30_at_9.54.43_AM.png) is ~$50,000,000 / yr, so call it 5¢ per year per customer. If your customer base is 5,000 companies then at 5¢ a pop you'd see $250 a year—an order of magnitude that fits with what was normal on Gittip. I guess you'd need 1000x—$50 per year per customer—to declare victory at $250,000 / yr in revenue. A consumer product like Wikipedia is going to have a lower price point—maybe 1000x lower, maybe not. They're also at the front edge of the curve in terms of the cultural shift to voluntarily paying for stuff, though of course we're not far behind (i.e., there's still plenty of shifting to do :).

Ultimately, no: I don't know if this is going to work or not. But it's fun! So I for one am going to keep chipping away at it. :-)

> But if gratipay's model is based on, "you need to go full time on your OSS project as a business, and make no money from anywhere else like a dayjob in order to sign up for us", that's not really a reasonable ask.

I sure hope it's not, because ...

> Most higher quality OSS projects are run by older folks who have dayjobs, families, mortgages who can't responsibly just chuck the whole thing to go full blown independent OSS vendor overnight. Most OSS projects are built by people who have day jobs.

I'm one of these folks. I've dropped back to half-time on Gratipay. In fact, _nobody_ working on Gratipay has gotten paid _at all_ from Gratipay since the [Gratipocalypse](https://medium.com/gratipay-blog/gratipocalypse-42fd0ec0d9e8) three months ago. Most of Gratipay's income (plus our reserves and some of my personal cash) has [gone to our lawyer](https://github.com/gratipay/inside.gratipay.com/issues/282). Yet [we're more active now than before](https://medium.com/gratipay-blog/hard-at-work-135000d1c237). Gratipay is fun and awesome and we're going to keep trying to get it off the ground. :airplane: 

There's no requirement that you can _only_ use Gratipay to fund a project, but the goal with Gratipay 2.0 is to build viable open businesses. That's not a _reality_ for any of the [61 approved teams](https://gratipay.com/explore/teams/) on Gratipay right now—we're only moving $1,400 altogether (~=$6,000/mo)—but if viability isn't at least the _goal_ then we're not on the same page.

Clearly, viability _is_ your goal, so I think we're probably actually on the same page. :-)

> I would use it to feed my family.

Sure. All of our teams/projects are currently so far below the viability level that it can seem like premature optimization to think about when we're _not_. But I think it's :chicken:-and-egg, because in order to get there together we need to collectively change our mindset: We are building viable open businesses. We are building viable open businesses. We are building viable open businesses. That's a new and hard thing to do! Let's dress for success! :dancer: :shipit:

**Comment by chadwhitacre** [comment]
@zzzeek By all means, try out Bountysource. They're probably about to steal our lunch, so you may as well go with the winner. :) But if you do still want to give Gratipay another shot, then I propose the following changes to the [SQLAlchemy team page](https://gratipay.com/sqlalchemy/):

> **Revenue Model**
> 
> Right now our funding comes from personal investment from core contributors. We want SQLAlchemy to be entirely sustained by voluntary payments from the many, many companies that depend on our software. We have a "Donate" button on our website and we remind people to contribute on Twitter.
> 
> **How Revenue is Shared**
> 
> Right now we have no revenue to share. Once the core developer is able to make a living from SQLAlchemy revenue we will share additional revenue with contributors.

I'm happy to make these changes for you since, as discussed, [we suck](https://github.com/gratipay/gratipay.com/issues/3545). Waddya say? :)

**Comment by zzzeek** [comment]
@whit537 those changes sound great, change away :).

**Comment by chadwhitacre** [comment]
Done and [approved](https://gratipay.com/sqlalchemy/)! Welcome aboard Gratipay 2.0, @zzzeek! :dancer:

**Comment by chadwhitacre** [comment]
@zzzeek And thank you. :-)

**Comment by mattbk** [comment]
🐢

On August 12, 2015 9:42:57 PM CDT, Chad Whitacre notifications@github.com wrote:

> @zzzeek And thank you. :-)
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/gratipay/applications/issues/11#issuecomment-130511519

## 

Sent from a phone that, although intelligent, is not street smart.

**Comment by chadwhitacre** [comment]
Hey @zzzeek, one last thing: you applied for the new team before we were requiring a PayPal account (see https://github.com/gratipay/gratipay.com/issues/3567), which means we have no way to get you your money, which means that we'll start processing payments for you again once you link your PayPal on [your Gratipay settings page](https://gratipay.com/about/me/settings/). Does PayPal work for you?

**Comment by zzzeek** [comment]
paypal is great.   you're able to process payments now?

**Comment by chadwhitacre** [comment]
> paypal is great.

Cool.

> you're able to process payments now?

Not sure exactly what you're asking. In general, we've been processing payments again since [May 21](https://medium.com/gratipay-blog/gratipay-2-0-2453d3c53077), but only for users who re-sign up using the team application. I had hoped to process payments for SQLAlchemy during [today's payday](https://github.com/gratipay/inside.gratipay.com/issues/310), but we didn't because you don't have a PayPal linked (along with six out of 69 others, it turns out).

Are you asking whether we are processing payments now _at all_? Or specifically whether we can try to squeeze you in for this week rather than waiting until next week to get you back on track?

**Comment by chadwhitacre** [comment]
> we didn't because you don't have a PayPal linked

... and since 2.0 we don't bring money _in_ unless we have a way to send it _out_ again. We had to drop ACH bank payouts when Balanced went under, so now we just offer PayPal. Once you link your PayPal address on [your settings page](https://gratipay.com/about/me/settings/) we will turn your 169 incoming payments back on. :-)

**Comment by zzzeek** [comment]
OK, that is done

**Comment by chadwhitacre** [comment]
Awesome, thank you. I'll keep an eye out on Thursday to make sure payments go through ...
